src/llama-context.cpp

```diff
@@ -112,11 +112,27 @@ llama_context::llama_context(
         }
     }
 
-    const uint32_t n_ctx_per_seq = cparams.n_ctx / cparams.n_seq_max;
+    if (cparams.kv_unified) {
+        cparams.n_ctx_seq = cparams.n_ctx;
+    } else {
+        cparams.n_ctx_seq = cparams.n_ctx / cparams.n_seq_max;
+    }
+
+    if (cparams.n_ctx_seq > hparams.n_ctx_train) {
+        LLAMA_LOG_WARN("%s: capping n_ctx_seq (%u) to n_ctx_train (%u)\n", __func__, cparams.n_ctx_seq, hparams.n_ctx_train);
+
+        cparams.n_ctx_seq = hparams.n_ctx_train;
+    }
+
+    if (cparams.kv_unified) {
+        cparams.n_ctx = cparams.n_ctx_seq;
+    } else {
+        cparams.n_ctx = cparams.n_ctx_seq * cparams.n_seq_max;
+    }
```

Comment:
I am not completely convinced about this, I think it may create confusion, and add complexity to applications. The server and other applications using the unified cache need a sequence length limit independent of `n_ctx`, but that should probably be a different parameter that defaults to `min(n_ctx, n_ctx_train)`. This would be an application parameter, not part of the llama.cpp API.

Reply:
Sounds good. Just remove the capping here?

Reply:
Yes, I think it would be preferable to not have a limit here. The user should be able to override the model `n_ctx_train`, and it is easier to do it this way than with a KV override.